doc/dune-files.rst

```diff
@@ -1635,6 +1635,28 @@ which for each ``g_mod`` in ``<mlg_list>`` is equivalent to:
      (deps (:mlg-file g_mod.mlg))
      (action (run coqpp %{mlg-file})))
 
+coq.extract
+-----------
+
+Coq may be instructed to *extract* OCaml sources as part of the compilation
+process. This is done using the ``coq.extract`` stanza:
+
+.. code:: lisp
+
+   (coq.extract
+    (prelude <name>)
+    (extracted_modules <names>)
+    <optional-fields>)
+
+- ``(prelude <name>)`` refers to the coq source that contains the extraction
+  commands.
+
+- ``(extraced_modules <names>)`` is an exhaustive list of OCaml modules
+  extracted.
+
+- ``<optional-fields>`` are ``flags``, ``theories``, and ``libraries``. All of
+  these fields have the same meaning as in the ``coq.theory`` stanza.
+
```

Comment:
IMHO it would be worth nothing that the extracted modules will become usable from a regular `(library ,...)` stanza; quite a few users of this feature will be new to Dune and this will not be obvious to them.

Reply:
Makes sense. By the way, what do you think of introducing `coq.rst` for all coq related documentation? That page would include a tutorial style introduction and a reference of all the coq related stanzas.

Reply:
:+1: 
It used to exist but was removed in the recent refactoring of the refman. But it would be great if it was reintroduced.

Reply:
Indeed it used to be its own, self-contained section; but was removed in favor of merging into the general stanza file. Personally I think it makes more sense the old way, so I'm all for it.

Old chapter can be seen here https://github.com/ocaml/dune/blob/1.11.4/doc/coq.rst

Reply:
I see. Was there a discussion regarding this change? I’d like to see the arguments for keeping everything in one place.